recipes/botan/all/conanfile.py

```diff
@@ -133,16 +108,44 @@ def config_options(self):
         if Version(self.version) >= "2.14.0":
             del self.options.single_amalgamation
 
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+        self._validate_compiler_settings()
+
+        if Version(self.version) >= "2.14.0":
+            self._validate_v2_14()
+
+        if self.options.get_safe("single_amalgamation"):
+            self.options.amalgamation = True
+
+        if self.options.with_boost:
+            self.options["boost"].add("shared=False")
+            self.options["boost"].add("magic_autolink=False")
+            self.options["boost"].add("without_coroutine=False")
+            self.options["boost"].add("without_system=False")
```

Comment:
What is this way to force dependencies options?

These values should be ckeched again in `validate()`

Reply:
I don't think this way of forcing values of options is documented...

Yes, better to check they have the proper value in the `validate()` method:

```python
def validate(self):
   if self.options["boost"]. magic_autolink == True:
      raise ConanInvalidConfiguration("Cannot use 'botan' with boost::magic_autolink because....)
```